### DISCUSSION TITLE
Expose prompting.

### DISCUSSION THREAD

**Discussion by mixelpixx** [cline/cline#1069] [Feature Requests]
Expose the **_prompting_** and **_tool use instructions_**, generate a way that we can have per model instructions if we desire, or need.  I like using Gemini, except it has issues with tool use and the diff editor, I feel like I could fix that myself with better tool explanation or prompting..

**Comment by mixelpixx** [comment]
issues like this with Gemini Flash 2.0, message from Gemini:

I'm still encountering issues with the _replace_in_file_ tool. It seems like the SEARCH block is not matching the content in src/rag_app.py, even though I'm using the content from the previous read_file output. I'll try reading the file again to make sure I have the correct content, and then I'll use write_to_file to correct the import statement.

**Comment by siscia** [comment]
This would be useful, indeed.

I was trying to use groq to understand if they llama code would have worked, but it just return an error. And I am not in the position to neither debug it myself, nor asking the provider what is wrong.

**Comment by slavakurilyak** [comment]
I'm seeing similar issues with `replace_in_file` while running the DeepSeek v3 model hosted with Hyperbolic

Exposing prompting and tool use instructions may help resolve these especially when paired with the newly released Auto-Approve mode
